# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/url-classifier/UrlClassifierFeatureEmailTrackingDataCollection.h
# Commit: d393067941bd
# Full Hash: d393067941bd69dfa7eff7790e19a0a0162076cd
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2022-07-20 09:59:35
# Regressor Bug: 1773701
# File Overlap Count: 1
# Description:
#   Bug 1773701 - Part 2: Implement the email tracking data collection feature. r=dimi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D151522
# ==============================================================================

diff -r f3c636e27b0b -r d393067941bd netwerk/url-classifier/UrlClassifierFeatureEmailTrackingDataCollection.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/url-classifier/UrlClassifierFeatureEmailTrackingDataCollection.h	Tue Jul 19 20:04:24 2022 +0000
@@ -0,0 +1,45 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_net_UrlClassifierFeatureEmailTrackingDataCollection_h
+#define mozilla_net_UrlClassifierFeatureEmailTrackingDataCollection_h
+
+#include "UrlClassifierFeatureBase.h"
+
+namespace mozilla::net {
+
+class UrlClassifierFeatureEmailTrackingDataCollection final
+    : public UrlClassifierFeatureAntiTrackingBase {
+ public:
+  static const char* Name();
+
+  static void MaybeShutdown();
+
+  static already_AddRefed<UrlClassifierFeatureEmailTrackingDataCollection>
+  MaybeCreate(nsIChannel* aChannel);
+
+  static already_AddRefed<nsIUrlClassifierFeature> GetIfNameMatches(
+      const nsACString& aName);
+
+  NS_IMETHOD ProcessChannel(nsIChannel* aChannel,
+                            const nsTArray<nsCString>& aList,
+                            const nsTArray<nsCString>& aHashes,
+                            bool* aShouldContinue) override;
+
+  NS_IMETHOD GetURIByListType(nsIChannel* aChannel,
+                              nsIUrlClassifierFeature::listType aListType,
+                              nsIUrlClassifierFeature::URIType* aURIType,
+                              nsIURI** aURI) override;
+
+ private:
+  UrlClassifierFeatureEmailTrackingDataCollection();
+
+  static void MaybeInitialize();
+};
+
+}  // namespace mozilla::net
+
+#endif  // mozilla_net_UrlClassifierFeatureEmailTrackingDataCollection_h